kuma/payments/forms.py

```diff
@@ -137,3 +197,137 @@ def make_charge(self):
                     )
                 )
         return False
+
+
+class RecurringPaymentForm(ContributionForm):
+
+    def __init__(self, *args, **kwargs):
+        super(RecurringPaymentForm, self).__init__(*args, **kwargs)
+        self.fields['donation_choices'].choices = RECURRING_PAYMENT__CHOICES
+
+    @staticmethod
+    def create_customer(email, token, user, name):
+        customer = stripe.Customer.create(
+            email=email,
+            source=token,
+            metadata={'name': name}
+        )
+        user.stripe_customer_id = customer.id
+        user.save()
+        return customer.id
+
+    @staticmethod
+    def update_source_name(source_id, name):
+        """Updates the source name with the users defined name"""
+        source = stripe.Source.retrieve(source_id)
+        source.owner["name"] = name
+        source.save()
+
+    def make_recurring_payment_charge(self, user, update_source_name=False):
+        amount = self.get_amount()
+        token = self.cleaned_data.get('stripe_token', '')
+        try:
+            if token and amount:
+                if user.stripe_customer_id:
+                    # ensure that customer is active
+                    customer = stripe.Customer.retrieve(user.stripe_customer_id)
+                    # if deleted make a new customer
+                    if 'deleted' in customer:
+                        customer_id = self.create_customer(
+                            self.cleaned_data['email'],
+                            token,
+                            user,
+                            self.cleaned_data['name']
+                        )
+                    else:
+                        customer_id = user.stripe_customer_id
+                else:
+                    customer_id = self.create_customer(
+                        self.cleaned_data['email'],
+                        token,
+                        user,
+                        self.cleaned_data['name']
+                    )
+
+                if update_source_name:
+                    self.update_source_name(
+                        token,
+                        self.cleaned_data['name']
+                    )
+                plan = stripe.Plan.create(
```

Comment:
I'd like to create a plan for each price, and reuse the same plan if it exists. Something like:

```python
plan_id = 'plan_monthly_%d_usd' % amount
try:
    plan = stripe.Plan.retrieve(plan_id)
except stripe.error.InvalidRequestError:
    plan = stripe.Plan.create(
        id=plan_id,
        amount=amount, 
        interval="month",
        product=settings.STRIPE_PRODUCT_ID,
        currency="usd",
    )
```

Reply:
yes this is much better solution 👍